src/base/static/js/views/place-form-view.js

```diff
@@ -1,136 +1,259 @@
-      this.bindCategoryListeners();
+      this.initializeDatetimePicker();
+      this.initializeRichTextFields();
+      this.setUrlTitlePrefix();
     },
-    bindCategoryListeners: function() {
-      $(".category-btn-container").off().on("click", function(evt) {
-        $(this).prev().trigger("click");
+
+    setUrlTitlePrefix: function() {
+      var self = this, 
+      layer = _.find(this.options.mapConfig.layers, function(layer) {
+        return layer.id === self.formState.selectedCategoryConfig.dataset;
       });
+
+      this.$el.find(".url-prefix")
+        .html(
+          window.location.protocol + "//" + window.location.hostname + "/" +
+          (layer.useSlugForCustomUrls ? layer.slug + "/" : "")
+        );
+    },
+
+    showCategorySeparator: function() {
+      this.$("#category-btns hr").show();
+    },
+
+    hideCategorySeparator: function() {
+      this.$("#category-btns hr").hide();
+    },
+
+    hideSilhouettes: function() {
+      this.$(".place-form-silhouettes").hide();
     },
-    checkAutocomplete: function() {
-      var self = this,
-      storedValue;
 
-      this.formState.selectedCategoryConfig.fields.forEach(function(field, i) {
-        storedValue = Util.getAutocompleteValue(field.name);
-        self.formState.selectedCategoryConfig.fields[i].autocompleteValue = storedValue || null;
+    resetFormState: function() {
+      this.formState = {
+        selectedCategoryConfig: {
+          fields: []
+        },
+        attachmentData: [],
+        commonFormElements: this.options.placeConfig.common_form_elements || {}
+      }
+    },
+
+    determineRenderabilityForEachCategory: function() {
+      _.each(this.options.placeConfig.place_detail, function(place) {
+        _.extend(place, {
+          isAdmin: Util.getAdminStatus(place.dataset),
+          isRenderable: function() {
+            if (!place.includeOnForm) {
+              return false;
+            } else if (place.admin_only && !Util.getAdminStatus(place.dataset)) {
+              return false;
+            }
+
+            return true;
+          }()
+        });
       });
+    },
+
+    determineFieldRenderability: function(categoryConfig, field) {
+      var renderability = {
+        isRenderable: true
+      }
+
+      if (field.admin_only && !categoryConfig.isAdmin) {
+        renderability.isRenderable = false;
+      }
+
+      return renderability;
+    },
+
+    // Before we render the fields for a given category, do the following:
+    // 1. Build an appropriate content object for each field
+    // 2. Check the autocomplete status of each field
+    // 3. Check the admin-only status of each field
+    prepareFormFieldsForRender: function() {
+      var self = this;
+
+      // Prepare category-specific fields
+      _.each(this.formState.selectedCategoryConfig.fields, function(field, i) { 
+        _.extend(this.formState.selectedCategoryConfig.fields[i],
+          Util.buildFieldContent(field, (field.autocomplete) ? 
```

Comment:
I've been trying to follow [jQuery style suggestions](https://contribute.jquery.org/style-guide/js/#multi-line-statements) for ternary expressions.